tools/data-api/internal/repositories/services.go

```diff
@@ -1,227 +1,411 @@
+	//								SegmentName:        "resourceGroupName",
+	//								ParsedFromParentID: false,
+	//							},
+	//						},
+	//					},
+	//					ReadMethod: MethodDefinition{
+	//						Generate:         true,
+	//						MethodName:       "VirtualMachineGet",
+	//						TimeoutInMinutes: 5,
+	//					},
+	//					Resource:        "VirtualMachine",
+	//					ResourceIdName:  "VirtualMachineId",
+	//					ResourceName:    "VirtualMachine",
+	//					SchemaModelName: "VirtualMachineSchema",
+	//					SchemaModels: map[string]TerraformSchemaModelDefinition{
+	//						"VirtualMachineSchema": {
+	//							Fields: map[string]TerraformSchemaFieldDefinition{
+	//								"VirtualMachineId": {
+	//									ObjectDefinition: TerraformSchemaFieldObjectDefinition{
+	//										NestedObject:  nil,
+	//										ReferenceName: nil,
+	//										Type:          StringTerraformSchemaFieldType,
+	//									},
+	//									Computed: false,
+	//									ForceNew: true,
+	//									HclName:  "virtual_machine_id",
+	//									Optional: false,
+	//									Required: true,
+	//									Documentation: TerraformSchemaDocumentationDefinition{
+	//										Markdown: "blah",
+	//									},
+	//									Validation: nil,
+	//								},
+	//								"Name": {
+	//									ObjectDefinition: TerraformSchemaFieldObjectDefinition{
+	//										NestedObject:  nil,
+	//										ReferenceName: nil,
+	//										Type:          StringTerraformSchemaFieldType,
+	//									},
+	//									Computed: false,
+	//									ForceNew: true,
+	//									HclName:  "name",
+	//									Optional: false,
+	//									Required: true,
+	//									Documentation: TerraformSchemaDocumentationDefinition{
+	//										Markdown: "blah",
+	//									},
+	//									Validation: nil,
+	//								},
+	//							},
+	//						},
+	//					},
+	//					Tests: TerraformResourceTestsDefinition{
+	//						BasicConfiguration:          "basictest",
+	//						RequiresImportConfiguration: "importtest",
+	//						CompleteConfiguration:       nil,
+	//						Generate:                    false,
+	//						OtherTests:                  nil,
+	//						TemplateConfiguration:       nil,
+	//					},
+	//					UpdateMethod: pointer.To(MethodDefinition{
+	//						Generate:         true,
+	//						MethodName:       "VirtualMachineCreateOrUpdate",
+	//						TimeoutInMinutes: 30,
+	//					}),
+	//				},
+	//			},
+	//		},
+	//
+	//		Generate: true,
+	//	},
+	//}, nil
 }
 
 func (s *ServicesRepositoryImpl) GetByName(serviceName string, serviceType ServiceType) (*ServiceDetails, error) {
-	allServices, err := s.GetAll(serviceType)
+	// GetByName builds the ServiceDetails for a singular service by calling processing functions to build the
+	// structs for the ServiceApiVersionDetails and ServiceApiVersionResourceDetails
+	serviceDirectory := fmt.Sprintf("%s/%s", s.directory, serviceName)
+	if _, err := os.Stat(serviceDirectory); os.IsNotExist(err) {
+		return nil, fmt.Errorf("service %s does not exist: %+v", serviceName, err)
```

Comment:
`%q` would be helpful here when there's whitespace (due to a typo/incorrect split) or similar

```suggestion
		return nil, fmt.Errorf("service %q does not exist: %+v", serviceName, err)
```